rust/src/lib.rs

```diff
@@ -219,21 +217,15 @@ impl Toodle {
 
     pub fn create_label(&mut self, name: String, color: String) -> Result<Option<Label>> {
         {
-            let mut in_progress = self.connection.begin_transaction()?;
-            let mut builder = TermBuilder::new();
-            let entid = builder.named_tempid("label".to_string());
+            let in_progress = self.connection.begin_transaction()?;
+            let mut builder = in_progress.builder().describe_tempid("label");
 
             let name_kw = kw!(:label/name);
-            let name_ref = in_progress.get_entid(&name_kw).ok_or_else(|| ErrorKind::UnknownAttribute(name_kw))?;
-            let _ = builder.add(entid.clone(), name_ref, TypedValue::typed_string(&name));
+            let _ = builder.add_kw(&name_kw, TypedValue::typed_string(&name));
```

Comment:
You use `name_kw` and `color_kw` only once, so you could inline them here and save a line.

Also you should tag a `?` on the end instead of `let _ =`.